token/program-2022/src/state.rs

```diff
@@ -295,7 +295,11 @@ const ACCOUNTTYPE_ACCOUNT: u8 = AccountType::Account as u8;
 impl GenericTokenAccount for Account {
     fn valid_account_data(account_data: &[u8]) -> bool {
         spl_token::state::Account::valid_account_data(account_data)
-            || ACCOUNTTYPE_ACCOUNT == *account_data.get(Account::LEN).unwrap_or(&0)
+            || (ACCOUNTTYPE_ACCOUNT
+                == *account_data
+                    .get(spl_token::state::Account::get_packed_len())
+                    .unwrap_or(&0)
+                && spl_token::state::is_initialized_account(account_data))
```

Comment:
There's the possibility that the provided account is actually a multisig, so you can add a length check for token-2022 here as:
```
account_data.len() >= Account::LEN && account_data.len() != Multisig::LEN
```

Reply:
Can `Account::LEN + SOME_EXTENSION_COMBO == Multisig::LEN` exist? :wink:

Reply:
It can when you specifically write tests for it :grin: https://github.com/solana-labs/solana-program-library/blob/d51b58294ff589a4e70b9b004e39bfbe6291e14d/token/program-2022/src/extension/mod.rs#L746-L749